ext/liquid_c/variable.c

```diff
@@ -128,8 +130,54 @@ void internal_variable_parse(variable_parse_args_t *parse_args)
     rb_rescue(try_variable_strict_parse, (VALUE)parse_args, variable_strict_parse_rescue, (VALUE)&rescue_args);
 }
 
+void internal_variable_compile(variable_parse_args_t *parse_args, unsigned int line_number)
+{
+    vm_assembler_t *code = parse_args->code;
+    vm_assembler_add_render_variable_rescue(code, line_number);
+    internal_variable_compile_evaluate(parse_args);
+    vm_assembler_add_pop_write(code);
+}
+
+static VALUE variable_strict_parse_method(VALUE self, VALUE markup)
+{
+    StringValue(markup);
+    check_utf8_encoding(markup, "markup");
+
+    VALUE parse_context = rb_ivar_get(self, id_ivar_parse_context);
+
+    expression_t *expression;
+    VALUE expression_obj = expression_new(&expression);
+
+    variable_parse_args_t parse_args = {
+        .markup = RSTRING_PTR(markup),
+        .markup_end = RSTRING_END(markup),
+        .code = &expression->code,
+        .code_obj = expression_obj,
+        .parse_context = parse_context,
+    };
+    try_variable_strict_parse((VALUE)&parse_args);
+    RB_GC_GUARD(markup);
```

Comment:
I'm not sure why this is needed. How is `markup` at risk of being GC'd?

Reply:
isn't needed before `try_variable_strict_parse` is called, so the `markup` local variable could get re-used by the compiler and remove the reference to the object from the C stack.  If the only reference to the `markup` object were passed into this method, then in theory, it seems like the GC could end up cleanup the string before we are done using it.

It probably isn't needed in practice, but it makes the code more obviously correct.

Reply:
Thanks for clarifying!